### PR TITLE
Logs

### PR DESCRIPTION
**Pull request by noahingham**
This PR adds two functions to the dark node delegate:

```go
SubscribeToLogs(chan do.Option)
UnsubscribeFromLogs(chan do.Option)
```

And the following function which calls the delegate:

```go
Logs(logRequest *rpc.LogRequest, stream rpc.DarkNode_LogsServer) error
```

### CODE REVIEW

File: node.go [hunk 1]
```diff
@@ -124,6 +126,29 @@ func (node *Node) SendOrderFragment(ctx context.Context, orderFragment *rpc.Orde
 	}
 }
 
+// Logs will create a logs channel with the delegate and send any received logs
+// through the RPC stream.
+func (node *Node) Logs(logRequest *rpc.LogRequest, stream rpc.DarkNode_LogsServer) error {
+	if node.Options.Debug >= DebugHigh {
+		log.Printf("[%v] received a log query", node.Address())
```
Comment (loongy) [inline node.go]: We don't use `[%v]` in the networking layer, instead use `%v`.

File: node.go [hunk 2]
```diff
@@ -303,3 +328,16 @@ func (node *Node) sync(syncRequest *rpc.SyncRequest, stream rpc.DarkNode_SyncSer
...
 }
+
+func (node *Node) logs(logsRequest *rpc.LogRequest, stream rpc.DarkNode_LogsServer) error {
+	logChannel := make(chan do.Option)
+	node.Delegate.SubscribeToLogs(logChannel)
+	defer node.Delegate.UnsubscribeFromLogs(logChannel)
+	for event := range logChannel {
+		// TODO: need to serialize data into the network representation
+		if err := stream.Send(event.Ok.(*rpc.LogEvent)); err != nil {
+			return err
```
Comment (loongy) [inline node.go]: Double checking, is unsubscribing from the logs sufficient? We don't need to co-ordinate the closure of the `logChannel`?

File: node_test.go [hunk 3]
```diff
@@ -286,34 +292,47 @@ var _ = Describe("dark network", func() {
...
+	// 		quit <- struct{}{}
+	// 		fmt.Println("quit, emptying...")
+	// 		for len(logs) > 0 {
+	// 			<-logs
+	// 		}
+	// 	})
+	// })
 	//
 	//	It("should be able to respond to sync query", func() {
 	//		option, quit := rpc.SyncWithTarget(to, from, DefaultTimeOut)
```
Comment (loongy) [inline node_test.go]: Why are all the tests commented out?

File: node.go (hunk 1 above)
Comment (noahingham) [reply node.go]: I'm leaving this as is to match the other logs in `node.go`

Comment (loongy) [reply node.go]: Cool. We'll do a larger review of how logging works soon.

**Pull request merged**